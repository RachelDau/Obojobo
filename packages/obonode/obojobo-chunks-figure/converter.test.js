import Converter from './converter'

describe('Figure Converter', () => {
	test('slateToObo converts a Slate node to an OboNode with content', () => {
		const slateNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {},
			children: [{ text: 'mockText', b: true }]
		}
		const oboNode = Converter.slateToObo(slateNode)

		expect(oboNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})

	test('oboToSlate converts an OboNode to a Slate node with caption, size and triggers', () => {
		const oboNode = {
			id: 'mockKey',
			type: 'mockType',
			content: {
				size: 'large',
				triggers: 'mock-triggers',
				textGroup: [
					{
						text: { value: 'mockCaption' }
					}
				]
			}
		}
		const slateNode = Converter.oboToSlate(oboNode)

		expect(slateNode).toMatchSnapshot()
	})
})
