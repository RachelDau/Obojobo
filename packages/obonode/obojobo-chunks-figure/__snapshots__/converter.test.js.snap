// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figure Converter oboToSlate converts an OboNode to a Slate node 1`] = `
Object {
  "children": Array [
    Object {
      "text": "",
    },
  ],
  "content": Object {},
  "id": "mockKey",
  "type": "mockType",
}
`;

exports[`Figure Converter oboToSlate converts an OboNode to a Slate node with caption, size and triggers 1`] = `
Object {
  "children": Array [
    Object {
      "text": "mockCaption",
    },
  ],
  "content": Object {
    "size": "large",
    "textGroup": Array [
      Object {
        "text": Object {
          "value": "mockCaption",
        },
      },
    ],
    "triggers": "mock-triggers",
  },
  "id": "mockKey",
  "type": "mockType",
}
`;

exports[`Figure Converter slateToObo converts a Slate node to an OboNode with content 1`] = `
Object {
  "children": Array [],
  "content": Object {
    "textGroup": Array [
      Object {
        "data": null,
        "text": Object {
          "styleList": Array [
            Object {
              "data": Object {},
              "end": 8,
              "start": 0,
              "type": "b",
            },
          ],
          "value": "mockText",
        },
      },
    ],
  },
  "id": "mockKey",
  "type": "mockType",
}
`;
